lib/spack/spack/detection/common.py

```diff
@@ -114,6 +114,11 @@ def is_executable(file_path):
     return os.path.isfile(file_path) and os.access(file_path, os.X_OK)
 
 
+def is_readable(file_path):
```

Comment:
I think `is_readable_file` is more appropriate (since this function excludes directories).

Also, this function isn't specifically related to detection, and so would likely be best off in `llnl.util.filesystem`.